src/wasm/wasm-binary.cpp

```diff
@@ -3560,40 +3560,40 @@ Result<> WasmBinaryReader::readInst() {
       return builder.makeLoad(8, false, offset, align, Type::f64, mem);
     }
     case BinaryConsts::I32StoreMem8: {
-      auto [mem, align, offset] = getMemarg();
-      return builder.makeStore(1, offset, align, Type::i32, mem);
+      auto [mem, align, offset, backing] = getMemargWithBacking();
```

Comment:
The binary format will need the type immediate as well, so this will have to look something like this:

```suggestion
      auto [mem, align, offset, backing] = getMemargWithBacking();
      if (backing == BackingType::Array) {
        HeapType type = getIndexedHeapType();
        ...
```

At that point we might as well add a separate `IRBuilder` method for making array stores, which should help separate concerns more.